Third task

stino-x opened the pull request:
Hello,

Iheagwara Augustine here, and I'd like to submit this pull request to resolve several tasks and queries related to Git and SQL. Below is a summary of the changes and tasks addressed in this pull request:

Git Changes:

Moved committed changes from one branch to another using git cherry-pick.
Demonstrated how to delete a row in a Git repository.
SQL Tasks:

Provided SQL queries to answer various questions related to an "animals" database, such as listing animals by owner, counting animals per species, and more.
Demonstrated SQL commands for modifying table structures, adding foreign keys, and altering primary keys.
Explained the use of transactions, savepoints, and committing changes in SQL.
Updated and inserted data into tables, including "owners" and "species."
Modified the "animals" table to include species and owner information.
Answered specific SQL queries related to the provided database.
These changes and queries aim to address the tasks and questions you presented. Please review them, and if everything looks good, feel free to merge this pull request.

Thank you for considering these changes, and please let me know if you have any questions or need further assistance.

Best regards,
Iheagwara Augustine

**IMAGES**
<img src=https://github.com/stino-x/vet-clinic-psql/assets/107426247/c49116c4-fea0-4728-b94e-4d7380ab01de>
<img src=https://github.com/stino-x/vet-clinic-psql/assets/107426247/4cf4b7de-edd5-428f-84b4-1c315b5c0fe3>
<img src=https://github.com/stino-x/vet-clinic-psql/assets/107426247/8d148ef6-7ba3-4a59-84e9-e8f2cab92f42>

stino-x committed:
Update database schema and add foreign key constraints (generated by useblackbox.ai )

stino-x committed:
Update animals table with species_id and owner_id based on name (generated by useblackbox.ai )

skyv26 reviewed (commented):
## Hi @stino-x, 

While you made a great effort in this project, unfortunately, I cannot proceed to review your code.

## Invalid Code Review Request 🟨 

![](https://media.giphy.com/media/v1.Y2lkPTc5MGI3NjExNGU0MmYyNmE3NGJmMGNmYTZjY2NkOWQ1NmIwM2I0MzdiYWJmZWEwNSZlcD12MV9pbnRlcm5hbF9naWZzX2dpZklkJmN0PWc/MShPkkzn6Scrd6VPK1/giphy.gif)


## Reason 🗒️ 
You have submitted a project, that doesn't have a professional `README` file because the `table of contents` is missing from your `README` file.

Your Code Review Request will be marked as invalid in your Dashboard, so please submit a new one once you are ready 🙏

#### Cheers, and Happy coding!👏👏👏

## ⚠️  WARNING ⚠️ 
**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

## Have any doubt ❓ 
Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag @skyv26 in your question so I can receive the notification. You can also follow me on the below platforms

<table>
<tr>
<td>

<a href="https://github.com/skyv26">
   <img src="https://img.shields.io/badge/GitHub-100000?style=for-the-badge&logo=github&logoColor=white" alt="@skyv26" />
</a>

</td>
<td>

<a href="https://www.linkedin.com/in/skyv2022/">
   <img src="https://img.shields.io/badge/LinkedIn-0077B5?style=for-the-badge&logo=linkedin&logoColor=white" alt="linkedin-skyv2022" />
</a>

</td>
<td>

<a href="https://twitter.com/vrma_aakash">
   <img src="https://img.shields.io/badge/Twitter-1DA1F2?style=for-the-badge&logo=twitter&logoColor=white" alt="@vrma_aakash" />
</a>

</td>
<td>

<a href="https://www.instagram.com/skybrel/">
   <img src="https://img.shields.io/badge/Instagram-E4405F?style=for-the-badge&logo=instagram&logoColor=white" alt="@skybrel" />
</a>

</td>
<td>

<a href="https://www.frontendmentor.io/profile/skyv26">
   <img src="https://user-images.githubusercontent.com/66476329/243158690-61eb1021-fbb6-464d-b841-a09c4caa156e.png" alt="@skybrel" />
</a>

</td>
</tr>
</table>

--------

_Invalid Code Review Request **does not count** into the code reviews limit._

stino-x committed:
Update README.md with table of contents (generated by useblackbox.ai )

Munsa1 commented on data.sql:
```diff
@@ -45,3 +45,23 @@ VALUES ('Blossom',  6,'1998-10-13', 17.0, true, 3, 'Grass-type');
 -- Inserting the seventh animal (Ditto)
 INSERT INTO animals (name, id, date_of_birth, weight_kg, neutered, escape_attempts, species)
 VALUES ('Ditto', 7, '2022-05-14', 22.0, true, 4, 'Normal-type');
+
+--- Task-4 ----
+
+-- Update animals to include species_id based on name
+UPDATE animals
+SET species_id = (CASE
+WHEN name LIKE '%mon' THEN (SELECT id FROM species WHERE name = 'Digimon')
+ELSE (SELECT id FROM species WHERE name = 'Pokemon')
+                 END);
+
+-- Update animals to include owner_id based on name
+UPDATE animals
+SET owner_id = (CASE
+WHEN name IN ('Agumon') THEN (SELECT id FROM owners WHERE full_name = 'Sam Smith')
+WHEN name IN ('Gabumon', 'Pikachu') THEN (SELECT id FROM owners WHERE full_name = 'Jennifer Orwell')
+WHEN name IN ('Devimon', 'Plantmon') THEN (SELECT id FROM owners WHERE full_name = 'Bob')
+WHEN name IN ('Charmander', 'Squirtle', 'Blossom') THEN (SELECT id FROM owners WHERE full_name = 'Melody Pond')
+WHEN name IN ('Angemon', 'Boarmon') THEN (SELECT id FROM owners WHERE full_name = 'Dean Winchester')
+                END);
```
- [ ] Kindly add the following data to the `data.sql` file for documentation purposes
```
 INSERT INTO owners (full_name, age)
 VALUES ('Sam Smith', 34);

 INSERT INTO owners (full_name, age)
 VALUES ('Jennifer Orwell', 19);

 INSERT INTO owners (full_name, age)
 VALUES ('Bob', 45);

 INSERT INTO owners (full_name, age)
 VALUES ('Melody Pond', 77);

 INSERT INTO owners (full_name, age)
 VALUES ('Dean Winchester', 14);

 INSERT INTO owners (full_name, age)
 VALUES ('Jodie Whittaker', 38);
```

Munsa1 reviewed (changes requested):
Hi @stino-x , 

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!
<FEEL FREE TO CHANGE THIS MESSAGE TO YOUR PERSONAL WORDS>

## Highlights
Passing Linters :heavy_check_mark: 


## Required Changes ♻️
_Check the comments under the review.
- [ ] Kindly add all screenshots for all queries as per requirements
![image](https://github.com/stino-x/vet-clinic-psql/assets/81932478/babeaaf2-91a7-4bda-8ded-cfbd32242c45)
![image](https://github.com/stino-x/vet-clinic-psql/assets/81932478/5b2e8413-3dd0-4f40-ac34-1fdfcc6c75f6)

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

You can also consider:
N/A

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

Munsa1 commented:
- [ ] Kindly remove the following broken links for a more descriptive readme file
![image](https://github.com/stino-x/vet-clinic-psql/assets/81932478/a84ed7c5-cbe5-4a91-b4c0-4ac19bfe63fa)
![image](https://github.com/stino-x/vet-clinic-psql/assets/81932478/3e048dae-fb0e-42a5-b8b0-299aa6314f7f)

stino-x committed:
Add owners and species tables and update animals table to include species_id (generated by useblackbox.ai )

laiifuu reviewed (changes requested):
# STATUS: CHANGES REQUIRED :recycle: 

## Hi @stino-x  :wave: 

Great work on making the changes requested by a previous reviewer 👏 
You've done well implementing some of the requested changes, but there are still some which aren't addressed yet.

![giphy](https://user-images.githubusercontent.com/93520178/261856992-3a2ad29c-e181-4e1b-bf50-a98ee69c28da.gif)

 ## Highlights
- [x] Data concerning the owners was added :+1: 

 ## Required Changes ♻️
- [ ] As mentioned by the previous reviewer, kindly remove the broken links in the table of contents of your README: 
    - Install
    - Run tests
    - Deployment

- [ ] As mentioned by the previous reviewer, all results of the queries should have a screenshot added to your PR description. Screenshots for the following queries are missing: 
    - What animals belong to Melody Pond
    - List of all animals that are Pokemon

## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._


Cheers and :computer: Happy coding :computer: 

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**


------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

stino-x commented:
@laiifuu  please check my screenshots again i did those 
What animals belong to Melody Pond
List of all animals that are Pokemon

stino-x committed:
Update README.md

This commit updates the prerequisites and usage sections of the README.md file. (generated by useblackbox.ai )

laiifuu commented:
Hey, I checked again and I don't see them: 

![grafitx_editor](https://github.com/stino-x/vet-clinic-psql/assets/93520178/69660976-5d69-436d-9c12-121a72911a9d)

![one](https://github.com/stino-x/vet-clinic-psql/assets/93520178/314c2cb6-8f2f-4676-8592-daaf316f7c78)

addod19 reviewed (approved):
## Hi @stino-x, 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 

### Highlights
- [ ] Great job adding the required screenshots :rocket: 


Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._
